week-4/D-methods/exercise-4.js

```diff
@@ -5,7 +5,10 @@ Define a method "makeFriend" to add a new friend to her list.
 
 var person = {
   name: "Alice",
-  friends: ["John", "Nina"]
+  friends: ["John", "Nina"],
+  makeFriend: function(name) {
+    this.friends[this.friends.length] = name;
```

Comment:
This is a case when I would use `push`. Indexing past the end of the array is a bit of a weird Javascript thing and best avoided. My advice would be to avoid using array indexes in general if you can think of another way to do something.